controllers/components/workbenches/workbenches_controller_actions.go

```diff
@@ -0,0 +1,151 @@
+package workbenches
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+)
+
+func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       NotebookControllerPath,
+		ContextDir: "",
+		SourcePath: "",
+	})
+	if err := odhdeploy.ApplyParams(
+		NotebookControllerPath,
+		nil,
+		map[string]string{
+			"namespace": rr.DSCI.Spec.ApplicationsNamespace,
+		},
+	); err != nil {
+		return fmt.Errorf("failed to update params.env from %s : %w", rr.Manifests[0], err)
+	}
+
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       KfNotebookControllerPath,
+		ContextDir: "",
+		SourcePath: "",
+	})
+	if err := odhdeploy.ApplyParams(
+		KfNotebookControllerPath,
+		nil,
+		map[string]string{
+			"namespace": rr.DSCI.Spec.ApplicationsNamespace,
+		},
+	); err != nil {
+		return fmt.Errorf("failed to update params.env from %s : %w", rr.Manifests[1], err)
+	}
+
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       NotebookImagesPath,
+		ContextDir: "",
+		SourcePath: "",
+	})
+	if err := odhdeploy.ApplyParams(
+		NotebookImagesPath,
+		nil,
+		map[string]string{
+			"namespace": rr.DSCI.Spec.ApplicationsNamespace,
+		},
+	); err != nil {
+		return fmt.Errorf("failed to update params.env from %s : %w", rr.Manifests[2], err)
+	}
```

Comment:
I am not sure to decide about this myself to be honest. Also, wouldn't the proposed removal also affect `devFlags()` action? Particularly the section between lines 111-125

Reply:
let's break down the function initialize()
the main purpose for this function is to set any extram values we want to pass into params.env
in workbenches case, i do not think we have such needed. in other components, some might need application namespace, or other value from flags.
so the part of append to rr.Manifests, is needed, but not the ApplyParams()

but i guess, if would like to replace the downstream image, it is doable for the NotebookControllerPath and KfNotebookControllerPath(my suggestion below) but not really required.

so in the most compacted version, i think this will do:
```
func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
		Path:       NotebookControllerPath,
		ContextDir: "",
		SourcePath: "",
	})
	
	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
		Path:       KfNotebookControllerPath,
		ContextDir: "",
		SourcePath: "",
	})
	
	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
		Path:       NotebookImagesPath,
		ContextDir: "",
		SourcePath: "",
	})
	return nil
}
```

Reply:
I see, thank you for clarification Wen! Changes were applied